### PR TITLE
Add mountNode prop to combos

### LINKED ISSUE #28625: [Feature]: Add `mountNode` property to Combobox
### Library

React Northstar / v0 (@fluentui/react-northstar)

### Describe the feature that you would like added

The Portal component's `mountNode` property lets the user specify the target DOM node into which the Portal/popover/flyout will render. This is a very useful property for cases where `<body>` or inline won't do, such as the PowerApps Canvas surface. 

The `InfoButton` component already has this available; it's exposed via the larger [`popover`](https://github.com/microsoft/fluentui/blob/99e835ade7328fcb47114f5d5d293c36c2665ae1/packages/react-components/react-infobutton/src/components/InfoButton/InfoButton.types.ts#L10) property. Something like that would be agreeable for Combobox, but I'll leave the implementation details up to the experts. 

### Have you discussed this feature with our team

GeoffCoxMSFT

### Additional context

This is an internal feature request from the PowerApps App fabric team.

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Issue

Allows callers to change the mountNode of the portal used to display the dropdown part for ComboBox and Dropdown.

Fixes #28625

## Changes
- Added mountNode to ComboboxBaseProps
- Updated Combobox and Dropdown to pass mountNode to Portal.

I created a storybook example to demonstrate it working.  I don't think this makes a good story to check in for combobox though.

https://github.com/microsoft/fluentui/assets/28762486/55186e61-be1f-4713-aa53-99bc295001fe

### CODE REVIEW

File: packages/react-components/react-combobox/src/components/Combobox/renderCombobox.tsx [hunk 1]
```diff
@@ -24,7 +23,7 @@ export const renderCombobox_unstable = (state: ComboboxState, contextValues: Com
           (state.inlinePopup ? (
             <slots.listbox {...slotProps.listbox} />
           ) : (
-            <Portal>
+            <Portal {...slotProps.portal}>
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-combobox/src/components/Combobox/renderCombobox.tsx:26]: I wanted to use slots.portal here, but Portal is a pure function and not a ForwardRef component, so I get this error:

JSX element type 'slots.portal' does not have any construct or call signatures.ts(2604)

File: packages/react-components/react-combobox/src/components/Dropdown/renderDropdown.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const renderDropdown_unstable = (state: DropdownState, contextValues: Dro
           (state.inlinePopup ? (
             <slots.listbox {...slotProps.listbox} />
           ) : (
-            <Portal>
+            <Portal {...slotProps.portal}>
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-combobox/src/components/Dropdown/renderDropdown.tsx:29]: I wanted to use slots.portal here, but Portal is a pure function and not a ForwardRef component, so I get this error:

JSX element type 'slots.portal' does not have any construct or call signatures.ts(2604)

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                 Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `87.985 kB`<br />`28.356 kB` | `88.049 kB`<br />`28.411 kB` | `64 B` <br />`55 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `86.388 kB`<br />`27.965 kB` | `86.466 kB`<br />`28.014 kB` | `78 B` <br />`49 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.576 kB`<br />`18.225 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `208.062 kB`<br />`57.968 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.409 kB`<br />`12.003 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |     `6.48 kB`<br />`2.203 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c6d2ace26c19b4c8102c77baf8f66ffeb36149f8'>c6d2ace26c19b4c8102c77baf8f66ffeb36149f8</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx [hunk 3]
```diff
@@ -184,6 +185,13 @@ export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLIn
         })
       : undefined;
 
+  const portalSlot: Slot<typeof Portal> | undefined =
+    open || hasFocus
+      ? resolveShorthand(props.portal, {
+          required: true,
+        })
+      : undefined;
```
Comment (smhigley) [inline packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx:193]: Unlike listbox, portal is always rendered, so would it make sense to remove the conditional?

File: packages/react-components/react-combobox/src/components/Combobox/Combobox.types.ts [hunk 4]
```diff
@@ -21,6 +22,11 @@ export type ComboboxSlots = {
 
   /* The dropdown listbox slot */
   listbox?: Slot<typeof Listbox>;
+
+  /* The dropdown portal slot.
+     Limited to setting props only as the listbox is the rendered child.
+  */
+  portal?: Slot<typeof Portal>;
```
Comment (smhigley) [inline packages/react-components/react-combobox/src/components/Combobox/Combobox.types.ts:29]: I think it'd make sense to have `portal` be non-optional and `NonNullable<>`, which then means you don't need the Omit/Pick below

**Review by smhigley — COMMENTED** [review]
Thanks so much for making this!

**Comment by layershifter** [comment]
Can we be consistent with other components and expose `mountNode` prop instead?

https://github.com/microsoft/fluentui/blob/9958feece96105b4dcc4c5965fb8fdd3987eae44/packages/react-components/react-tooltip/src/components/Tooltip/Tooltip.types.ts#L38
https://github.com/microsoft/fluentui/blob/9958feece96105b4dcc4c5965fb8fdd3987eae44/packages/react-components/react-menu/src/components/Menu/Menu.types.ts#L15
https://github.com/microsoft/fluentui/blob/9958feece96105b4dcc4c5965fb8fdd3987eae44/packages/react-components/react-popover/src/components/Popover/Popover.types.ts#L17

**Review by smhigley — APPROVED** [review]

**Comment by GeoffCoxMSFT** [comment]
> Can we be consistent with other components and expose `mountNode` prop instead?
> 
> https://github.com/microsoft/fluentui/blob/9958feece96105b4dcc4c5965fb8fdd3987eae44/packages/react-components/react-tooltip/src/components/Tooltip/Tooltip.types.ts#L38
> 
> 
> https://github.com/microsoft/fluentui/blob/9958feece96105b4dcc4c5965fb8fdd3987eae44/packages/react-components/react-menu/src/components/Menu/Menu.types.ts#L15
> 
> 
> https://github.com/microsoft/fluentui/blob/9958feece96105b4dcc4c5965fb8fdd3987eae44/packages/react-components/react-popover/src/components/Popover/Popover.types.ts#L17

That was my original plan, then others suggested making it a slot. LOL.  I will update to follow the precedent.

**Pull request merged**